src/cmap/auth/mongodb_oidc/aws_device_workflow.ts

```diff
@@ -0,0 +1,27 @@
+import { readFile } from 'fs/promises';
+
+import { MongoAWSError } from '../../../error';
+import { DeviceWorkflow } from './device_workflow';
+
+/**
+ * Device workflow implementation for AWS.
+ *
+ * @internal
+ */
+export class AwsDeviceWorkflow extends DeviceWorkflow {
+  constructor() {
+    super();
+  }
+
+  /**
+   * Get the token from the environment.
+   */
+  async getToken(): Promise<string> {
+    const tokenFile = process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+    if (tokenFile) {
+      return readFile(tokenFile, 'utf8');
+    } else {
```

Comment:
technically, the else wrapper isn't necessary here since L22 returns; I recommend `if (!tokenFile) { throw... }` and then `return readFile` on the bottom so that we follow a pattern of "validate then execute" as we do in the majority of the codebase

Reply:
Inverted.